azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java

```diff
@@ -348,6 +363,21 @@ public WebApp getWebApp() throws AzureAuthFailureException {
         return null;
     }
 
+    public DeploymentSlot getDeploymentSlot(final WebApp app, final String slotName) {
+        DeploymentSlot slot = null;
+        if (!StringUtils.isEmpty(slotName)) {
+            try {
+                slot = app.deploymentSlots().getByName(slotName);
+            } catch (NoSuchElementException deploymentSlotNotExistException) {

```

Comment:
swallow the deploymentSlotNotExistException is fine here. But we need to differentiate:
- user doesn't set these configuration
- user set but the slot name is invalid